packages/farrow-schema/src/validator.ts

```diff
@@ -285,8 +285,13 @@ Validator.impl<S.StructType>(S.StructType, (schema) => {
 
       for (const key in fields) {
         const Field = fields[key]
+        const schema = Field[S.Type]
+        if (schema === S.Never && !(key in input)) {
```

Comment:
Why handle `Never` as a special case here?

There is no semantics to ignore a field of an object with `Never type`. 

```ts
class Test extends ObjectType {
  a = Number
  b ?= Nullable(Never) // serve as reserve field to disallow any subtype with b.
}
```

The usage of `Never` for reserve fields should combine `Nullable` type.

Reply:
In this case

```ts
const validateStructWithNever = createSchemaValidator(Struct({ foo: Number, bar: Never }))
expect(assertOk(validateStructWithNever({ foo: 0 }))).toEqual({ foo: 0 })
```
if donot handle never as special case, there is a `bar` key in `fields`, so it must run until `Validation.validate(Never, undefined)`, will throw error, but it is not expected.

Reply:
The code-gen type of Never does not work as you mentioned above.
```ts
type Test = {
  foo: number,
  bar: never
}

const test = (arg: Test) => {}

test({
  foo: 1,
  bar: undefined // this can't type-check
})
```